File: 560. Subarray Sum Equals K/subarraySum.py
from typing import List

class Solution:
  def subarraySum(self, nums: List[int], k: int) -> int:
    count = 0
    currSum = 0
    prefixSum = { 0: 1 }

    for n in nums:
      currSum += n
      diff = currSum - k
      if diff in prefixSum:
        count += prefixSum[diff]
      prefixSum[currSum] = prefixSum.get(currSum, 0) + 1
    
    return count
  

nums = [1, 1, 1]
k = 2
solution = Solution()
print('subarraySum: ', solution.subarraySum(nums, k)) # 2

File: 560. Subarray Sum Equals K/subarraySum.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
const subarraySum = (nums, k) => {
  let count = 0;
  let currSum = 0;
  const prefixSum = { 0: 1 };
  for (let i = 0; i < nums.length; i++) {
    currSum += nums[i];
    const diff = currSum - k;
    if (prefixSum[diff]) {
      count += prefixSum[diff];
    }
    prefixSum[currSum] = (prefixSum[currSum] || 0) + 1;
  }

  return count;
};

const nums = [1, 1, 1];
const k = 2;
console.log("subarraySum: ", subarraySum(nums, k)); // 2
